server/service/integration_mdm_dep_test.go

```diff
@@ -3179,3 +3180,307 @@ func (s *integrationMDMTestSuite) TestEnforceMiniumOSVersion() {
 		})
 	})
 }
+
+func (s *integrationMDMTestSuite) TestSetupExperienceVPPInstallError() {
+	t := s.T()
+	ctx := context.Background()
+
+	teamDevice, enrolledHost, team := s.createTeamDeviceForSetupExperienceWithProfileSoftwareAndScript()
+
+	orgName := "Fleet Device Management Inc."
+	token := "mycooltoken"
+	expTime := time.Now().Add(200 * time.Hour).UTC().Round(time.Second)
+	expDate := expTime.Format(fleet.VPPTimeFormat)
+	tokenJSON := fmt.Sprintf(`{"expDate":"%s","token":"%s","orgName":"%s"}`, expDate, token, orgName)
+	t.Setenv("FLEET_DEV_VPP_URL", s.appleVPPConfigSrv.URL)
+	var validToken uploadVPPTokenResponse
+	s.uploadDataViaForm("/api/latest/fleet/vpp_tokens", "token", "token.vpptoken", []byte(base64.StdEncoding.EncodeToString([]byte(tokenJSON))), http.StatusAccepted, "", &validToken)
+
+	var getVPPTokenResp getVPPTokensResponse
+	s.DoJSON("GET", "/api/latest/fleet/vpp_tokens", &getVPPTokensRequest{}, http.StatusOK, &getVPPTokenResp)
+
+	// Add an app with 0 licenses available
+	s.appleVPPConfigSrvConfig.Assets = append(s.appleVPPConfigSrvConfig.Assets, vpp.Asset{
+		AdamID:         "5",
+		PricingParam:   "STDQ",
+		AvailableCount: 0,
+	})
+
+	t.Cleanup(func() {
+		s.appleVPPConfigSrvConfig.Assets = []vpp.Asset{
```

Comment:
Can we extract this declaration and the one in suite setup so they're guaranteed to have the same value? We already have drift on these assets (and maybe should resolve that now) and having a fourth place where things are reset means that we're one suite change away from _other_ tests behaving differently in isolation than in-suite.

Reply:
Ya good point!